<?xml version="1.0" encoding="UTF-8"?><rss version="2.0"
	xmlns:content="http://purl.org/rss/1.0/modules/content/"
	xmlns:dc="http://purl.org/dc/elements/1.1/"
	xmlns:atom="http://www.w3.org/2005/Atom"
	xmlns:sy="http://purl.org/rss/1.0/modules/syndication/"
	
	>
<channel>
	<title>Comments on: Catch block and type conversion in C++</title>
	<atom:link href="http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/feed/" rel="self" type="application/rss+xml" />
	<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/</link>
	<description>A computer science portal for geeks</description>
	<lastBuildDate>Tue, 17 Nov 2015 05:17:31 +0000</lastBuildDate>
	<sy:updatePeriod>hourly</sy:updatePeriod>
	<sy:updateFrequency>1</sy:updateFrequency>
	<generator>https://wordpress.org/?v=4.6.8</generator>
	<item>
		<title>By: Gallon</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-33841</link>
		<dc:creator><![CDATA[Gallon]]></dc:creator>
		<pubDate>Tue, 25 Mar 2014 18:48:00 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-33841</guid>
		<description><![CDATA[I think in 1st example the reason it doesn&#039;t catch &#039;x&#039; is because it&#039;s a char type in C++ (instead of integer type in C)]]></description>
		<content:encoded><![CDATA[<p>I think in 1st example the reason it doesn&#8217;t catch &#8216;x&#8217; is because it&#8217;s a char type in C++ (instead of integer type in C)</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: GeeksforGeeks</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-18852</link>
		<dc:creator><![CDATA[GeeksforGeeks]]></dc:creator>
		<pubDate>Wed, 29 May 2013 18:17:28 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-18852</guid>
		<description><![CDATA[Thanks for pointing this out.  We have updated the code.  Keep it up!]]></description>
		<content:encoded><![CDATA[<p>Thanks for pointing this out.  We have updated the code.  Keep it up!</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: kewl_coder</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-18851</link>
		<dc:creator><![CDATA[kewl_coder]]></dc:creator>
		<pubDate>Wed, 29 May 2013 18:12:18 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-18851</guid>
		<description><![CDATA[Here, there is a fundamental flaw(though minor) in the code which is getting neglected due to the test case followed. 
when we do &quot;throw new MyExcept;&quot;, then the corresponding catch block should have a pointer i.e. catch(MyExcept1 *e) or catch(MyExcept1 *e) whichever you want.]]></description>
		<content:encoded><![CDATA[<p>Here, there is a fundamental flaw(though minor) in the code which is getting neglected due to the test case followed.<br />
when we do &#8220;throw new MyExcept;&#8221;, then the corresponding catch block should have a pointer i.e. catch(MyExcept1 *e) or catch(MyExcept1 *e) whichever you want.</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Ish</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-7097</link>
		<dc:creator><![CDATA[Ish]]></dc:creator>
		<pubDate>Sun, 15 Jan 2012 06:35:54 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-7097</guid>
		<description><![CDATA[Thanks !]]></description>
		<content:encoded><![CDATA[<p>Thanks !</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: kartik</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-7096</link>
		<dc:creator><![CDATA[kartik]]></dc:creator>
		<pubDate>Sun, 15 Jan 2012 05:27:41 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-7096</guid>
		<description><![CDATA[@Ish: yes, the same type and ancestors type (http://www.geeksforgeeks.org/archives/9279)]]></description>
		<content:encoded><![CDATA[<p>@Ish: yes, the same type and ancestors type (<a href="http://www.geeksforgeeks.org/archives/9279" rel="nofollow">http://www.geeksforgeeks.org/archives/9279</a>)</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Ish</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-7076</link>
		<dc:creator><![CDATA[Ish]]></dc:creator>
		<pubDate>Fri, 13 Jan 2012 04:59:49 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-7076</guid>
		<description><![CDATA[Does this example tell us that a catch block will accept only the values of the type have been mentioned with the catch block ?]]></description>
		<content:encoded><![CDATA[<p>Does this example tell us that a catch block will accept only the values of the type have been mentioned with the catch block ?</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: kartik</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-6992</link>
		<dc:creator><![CDATA[kartik]]></dc:creator>
		<pubDate>Thu, 29 Dec 2011 06:52:22 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-6992</guid>
		<description><![CDATA[@yash:  A single paramete constructor works as a conversion constructor in C++.  For example, the following prgram prints

&lt;pre&gt;
Conversion constructor called
fun called
&lt;/pre&gt;

&lt;pre&gt; &lt;code language=&quot;C&quot;&gt;
#include &#060;iostream&#062;
using namespace std;

class MyExcept1 {};

class MyExcept2
{
public:

    // Conversion constructor
    MyExcept2 (const MyExcept1 &#038;e1 )
    {
        cout &#060;&#060; &#034;Conversion constructor called&#034; &#060;&#060; endl;
    }
};

void fun(MyExcept2 e2)
{
    cout &#060;&#060; &#034;fun called&#034; &#060;&#060; endl;
}

int main()
{
    MyExcept1 e1;
    fun(e1);
    return 0;
}
&lt;/code&gt; &lt;/pre&gt;

The example given in post is to point that such conversion doesn&#039;t happen for catch blocks.]]></description>
		<content:encoded><![CDATA[<p>@yash:  A single paramete constructor works as a conversion constructor in C++.  For example, the following prgram prints</p>
<pre>
Conversion constructor called
fun called
</pre>
<pre> <code language="C">
#include &lt;iostream&gt;
using namespace std;

class MyExcept1 {};

class MyExcept2
{
public:

    // Conversion constructor
    MyExcept2 (const MyExcept1 &amp;e1 )
    {
        cout &lt;&lt; &quot;Conversion constructor called&quot; &lt;&lt; endl;
    }
};

void fun(MyExcept2 e2)
{
    cout &lt;&lt; &quot;fun called&quot; &lt;&lt; endl;
}

int main()
{
    MyExcept1 e1;
    fun(e1);
    return 0;
}
</code> </pre>
<p>The example given in post is to point that such conversion doesn&#8217;t happen for catch blocks.</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: yash</title>
		<link>http://www.geeksforgeeks.org/catch-block-and-type-conversion-in-c/#comment-6989</link>
		<dc:creator><![CDATA[yash]]></dc:creator>
		<pubDate>Thu, 29 Dec 2011 00:51:38 +0000</pubDate>
		<guid isPermaLink="false">http://www.cdn.geeksforgeeks.org/?p=16432#comment-6989</guid>
		<description><![CDATA[i dont exactly understand the second example]]></description>
		<content:encoded><![CDATA[<p>i dont exactly understand the second example</p>
]]></content:encoded>
	</item>
</channel>
</rss>

<!-- Dynamic page generated in 0.040 seconds. -->
<!-- Cached page generated by WP-Super-Cache on 2017-11-24 13:19:42 -->

<!-- Compression = gzip -->